h/browser/chrome/lib/sidebar-injector.js

```diff
@@ -288,6 +308,22 @@ function SidebarInjector(chromeTabs, dependencies) {
   function injectScript(tabId, path) {
     return executeScriptFn(tabId, {file: path});
   }
+
+  /**
+   * Inject configuration information for the Hypothesis application
+   * into the page as JSON data via a <meta> tag.
+   *
+   * A <meta> tag is used because that makes it available to JS content
+   * running in isolated worlds.
+   */
+  function injectConfig(tabId, config) {
```

Comment:
One difference is that these are a different set of settings. `js-hypothesis-settings` in `app.html` are the sidebar's private settings. This is another way to specify the same set of public settings as `window.hypothesisConfig`. A `<meta>` tag seemed the obvious way to do a key/value pair to me, although we could perhaps use a script tag with `<script class="js-hypothesis-config">` instead?

I think the biggest reason to use one vs. the other would be if there was some environment where using one or the other caused a website compatibility problem.